docs/application/native/guides/compliations/watchface-complication.md

```diff
@@ -0,0 +1,341 @@
+| watchface_complication_data_get_extra_data() | extra data |
+
+```cpp
+void _watchface_complication_updated_cb(int complication_id,
+		const char *provider_id,
+		watchface_complication_type_e type,
+		const bundle *data, void *user_data)
+{
+	char *shorttext;
+	if (type == WATCHFACE_COMPLICATION_TYPE_SHORT_TEXT)
+		watchface_complication_data_get_short_text(data, &shorttext);
+}
+```
+
+
+If the callback is not used any more, it should be be removed by using watchface_complication_remove_updated_cb function.
+
+```cpp
+{
+	watchface_complication_remove_updated_cb(complication,
+		_watchface_complication_updated_cb);
+}
+```
+
+
+If the watchface want to get the updated complication data, watchface can request to update the data to the complication provider. 
+
+```cpp
+{
+	watchface_complication_send_update_request(complication);
+}
+```
+
+
+# Specify to support only specific complication providers
+
+Watchface can support the specific complication provider by using allowed_list.
+In This case, watchface supports complications only in allowed complication list.
+
+```cpp
+{
+	complication_allowed_list_h allowed_list;
+
+	watchface_complication_allowed_list_create(&allowed_list); 
+	watchface_complication_allowed_list_add(allowed_list, "PROVIDER_ID_1", 
+			WATCHFACE_COMPLICATION_TYPE_SHORT_TEXT); 
+	watchface_complication_allowed_list_add(allowed_list, "PROVIDER_ID_2", 
+			WATCHFACE_COMPLICATION_TYPE_SHORT_TEXT); 
+	watchface_complication_allowed_list_apply(complication, allowed_list);
+
+	watchface_complication_allowed_list_destroy(allowed_list);
+}
+```
+ 
+ 
+
+# Watchface Edit with Editable feature
+
+The watchface application can provide the changeable features(compliation, color, font..) with watchface editable APIs.
+The features can be changed by watchface editor application.
+
+
+![Complication_editor](media/editor_1.png)
+
+
+To use the functions and data types for the watchface_editable API, include the <watchface-editable.h>.
+
+```cpp
+#include <watchface-editable.h>
+```
+ 
+1. When the editor is ready to edit, it notifies the edit ready. To receive the notification from the editor, watchface_editable_edit_ready_cb() should
+be added.
+
+```cpp
+void _watchface_editable_edit_ready_cb(watchface_editable_container_h handle,
+		const char *editor_appid,
+		void *user_data)
+{
+}
+
+{
+	watchface_editable_add_edit_ready_cb(_watchface_editable_edit_ready_cb, NULL);
+}
+```
+ 
+2. Watchface can make the complication_candidates_list_h to identify the editable features to request to editor.
+Watchface can request to edit the features by using watchface_editable_request_edit().
+
+3. The watchface editor can draw highlight to give the information what is editing.
+To inform the position of editable features, watchface_editable_highlight_h should be used.
+The highlight handle can be created by watchface_editable_hilight_create() function with watchface shape type.
+
+There are following shape types.
+
+| Shape types | Description |

```

Comment:
For the indentation, please add 4 spaces in front of the line from 235 to 238, and line 241~248.